### PR TITLE
Database setup

### PR DESCRIPTION
**Pull request by DarkPurple141**
Merge to get master up to date with most recent build.

DB not really finished. Added some tests for sanity.

### CODE REVIEW

File: model/user.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Game'
+  }, // if a logout occurs can fetch gameID
+  admin: Boolean,
+  created_at: {
+    type: Date,
+    default: Date.now
+  },
+  updated_at: Date
+  /* Things we don't strictly need atm
```
Comment (zainafzal08) [inline model/user.js]: should we just remove this for now?

Comment (zainafzal08) [reply model/user.js]: @DarkPurple141

Comment (DarkPurple141) [reply model/user.js]: @zainafzal08 remove cos the update didn't seem to work for you. Bad Zain. Bad Zain. Sadly a pain I suffered myself yesterday, the data object returned by update is the old data object. Thus your assertion was failing.

Comment (DarkPurple141) [reply model/user.js]: The way to avoid this according the docs is add the option { new : true }

**Review by zainafzal08 — APPROVED** [review]
Looking good!

**Pull request merged**